src/modules/musig/session_impl.h

```diff
@@ -418,13 +493,20 @@ static int secp256k1_musig_nonce_process_internal(int *fin_nonce_parity, unsigne
     }
     /* fin_nonce = aggnonce[0] + b*aggnonce[1] */
     secp256k1_scalar_set_b32(b, noncehash, NULL);
-    secp256k1_ecmult(&fin_nonce_ptj, &aggnoncej[1], b, NULL);
-    secp256k1_gej_add_ge(&fin_nonce_ptj, &fin_nonce_ptj, &aggnonce[0]);
+    secp256k1_gej_set_infinity(&fin_nonce_ptj);
+    if (!secp256k1_gej_is_infinity(&aggnoncej[1])) {
+        secp256k1_ecmult(&fin_nonce_ptj, &aggnoncej[1], b, NULL);
+    }
+    if (!secp256k1_gej_is_infinity(&aggnoncej[0])) {
+        secp256k1_gej_add_ge(&fin_nonce_ptj, &fin_nonce_ptj, &aggnonce[0]);
+    }
```

Comment:
I would be surprised if `secp256k1_ecmult` or `secp256k1_gej_add_ge` can't handle infinity as an input. Or is this just supposed to be an optimization? If yes, I suggest to drop it because it optimizes the malicious case.

Reply:
Ah thanks, I wanted to revisit this. `secp256k1_gej_add_ge` can't deal with infinity, but `secp256k1_gej_add_ge_var` can. Fixed.